src/studio/src/designer/backend/Controllers/ModelController.cs

```diff
@@ -99,27 +100,32 @@ public ActionResult Upload(string org, string app, IFormFile thefile)
             return Json(false);
         }
 
-        private void HandleTexts(string org, string app, Dictionary<string, Dictionary<string, string>> allTexts)
+        private void HandleTexts(string org, string app, Dictionary<string, Dictionary<string, TextResourceElement>> modelTexts)
         {
-            Dictionary<string, Dictionary<string, string>> existingTexts = _repository.GetServiceTexts(org, app);
+            // <textResourceElement.Id <language, textResourceElement>>
+            Dictionary<string, Dictionary<string, TextResourceElement>> existingTexts = _repository.GetServiceTexts(org, app);
 
             if (existingTexts == null)
             {
-                existingTexts = new Dictionary<string, Dictionary<string, string>>();
+                existingTexts = new Dictionary<string, Dictionary<string, TextResourceElement>>();
             }
 
-            foreach (KeyValuePair<string, Dictionary<string, string>> cultureString in allTexts)
+            foreach (KeyValuePair<string, Dictionary<string, TextResourceElement>> item in modelTexts)
```

Comment:
`item` can be renamed to textResource.

Reply:
Tried going with textResourceElementDict.. A complex object so was hard to find a good descriptive name